### PR TITLE
OnlineDDL: mitigate scenario where a migration sees recurring cut-over timeouts

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Description

Issue details in https://github.com/vitessio/vitess/issues/12450

This PR offers a quick mitigation like so:

- Increase of the `const` value of `vreplicationCutOverThreshold` to `10sec`. The absolute value of `5sec` is just too low for the combined delays incurred by even the smallest replication lag, plus wait interval times. It's too marginal.

In a future PR we will make the threshold dynamic, and collaborative with the throttler's threshold, if enabled. But in the meantime `10sec` is a substantial improvement for most situations.

Additionally, this PR now `DROP`s the temporary sentry table used in a cut-over step, as soon as the cut-over is completed (or failed), so that we don't get a proliferation of tables in the schema. If the table is successfully dropped, then we also remove it from the `artifacts` column, so that the column does not get bloated. If, for any reason, the table cannot be dropped, we still complete the cut-over step, and we let the existing `gcartifacts()` and table GC mechanisms to eventually reclaim the table. Noteworthy that the sentry table is empty, and never populated, and so `DROP`ping it is a fast and safe operation.

## Related Issue(s)

- Fixes https://github.com/vitessio/vitess/issues/12450
- https://github.com/vitessio/vitess/issues/6926

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [ ] Tests were added or are not required
-   [ ] Documentation was added or is not required

### CODE REVIEW

**Review by notfelineit — APPROVED** [review]
the in-code comments are so great!

**Comment by shlomi-noach** [comment]
`onlineddl_vrepl_stress` is failing and probably related. Checking.

**Comment by shlomi-noach** [comment]
Ping for review

File: go/vt/vttablet/tabletserver/query_executor.go [hunk 1]
```diff
@@ -66,7 +66,7 @@ type QueryExecutor struct {
 
 const (
 	streamRowsSize         = 256
-	maxQueryBufferDuration = 10 * time.Second
+	maxQueryBufferDuration = 15 * time.Second
```
Comment (shlomi-noach) [inline go/vt/vttablet/tabletserver/query_executor.go:69]: This is the single most important change in this PR, an increase in timeout.

File: go/vt/vttablet/onlineddl/executor.go [hunk 2]
```diff
@@ -138,7 +138,7 @@ const (
 	emptyHint                                = ""
 	readyToCompleteHint                      = "ready_to_complete"
 	databasePoolSize                         = 3
-	vreplicationCutOverThreshold             = 5 * time.Second
+	vreplicationCutOverThreshold             = 10 * time.Second
```
Comment (shlomi-noach) [inline go/vt/vttablet/onlineddl/executor.go:141]: This is the single most important change in this PR, an increase in timeout.

**Pull request merged**